integration/apiserver.go

```diff
@@ -24,6 +24,18 @@ type APIServer struct {
 	// doc.go) for details.
 	Path string
 
+	// Args is a list of arguments which will passed to the APIServer binary.
```

Comment:
This seems a little bit more convoluted than we might want. But at the same time you only use "Args" and even the templates if you really have to, so I guess it's fine.

Reply:
I see that. I had some different ideas like:
- Have a set of "static" arguments, that a user can override (a simple `[]string`). The "dynamic" arguments (e.g. `--data-dir=/some/dir`) will be generated in any case and pushed on top of the arguments list.
- Have a function on `APIServer` / `Etcd` like `func (dir String, listenURL *url.URL) (generatedArgs []string)` which can be specified / overridden by a user.
- ... some more ...

None of the ideas got me really excited. Either they solved just parts of the problem ("dynamic" vs. "static" arguments) or where changing the interface in a somewhat weird way (by adding some strange function to `Etcd`/`APIServer` and potentially specify some new `Interface`).
So that's the best I could come up with ... not saying it is the best or only solution, however.

As you said, a user has now the options to
- do nothing and use the defaulting mechanism
- override the args with simple strings
- or override (parts of) the args with template strings and by doing so, have access to certain computed/defaultes values of the struct

Also, now thinking about this, I think I will add some documentation about arguments in `doc.go` and maybe give some example use-cases.